### PR TITLE
get and has method else removed

### PR DESCRIPTION
**Pull request by khanrn**
`get` and `has` method's redundant `else` condition removed.

### CODE REVIEW

**Comment by browner12** [comment]
this changes execution since the `if` condition is not returning anything.

File: src/Illuminate/Support/Arr.php [hunk 1]
```diff
@@ -294,11 +294,11 @@ public static function get($array, $key, $default = null)
         }
 
         foreach (explode('.', $key) as $segment) {
-            if (static::accessible($array) && static::exists($array, $segment)) {
-                $array = $array[$segment];
-            } else {
-                return value($default);
+            if (!static::accessible($array) && !static::exists($array, $segment)) {
```
Comment (nunomaduro) [inline src/Illuminate/Support/Arr.php]: I believe you made a mistake here. The code you are looking for is:

```
if (! static::accessible($array) ||  ! static::exists($array, $segment)) {
```

File: src/Illuminate/Support/Arr.php [hunk 2]
```diff
@@ -335,11 +335,11 @@ public static function has($array, $keys)
             }
 
             foreach (explode('.', $key) as $segment) {
-                if (static::accessible($subKeyArray) && static::exists($subKeyArray, $segment)) {
-                    $subKeyArray = $subKeyArray[$segment];
-                } else {
-                    return false;
+                if (!static::accessible($subKeyArray) && !static::exists($subKeyArray, $segment)) {
```
Comment (nunomaduro) [inline src/Illuminate/Support/Arr.php]: I believe you made a mistake here. The code you are looking for is:
```
if (! static::accessible($subKeyArray) || ! static::exists($subKeyArray, $segment)) {
```

**Review by nunomaduro — CHANGES_REQUESTED** [review]
I believe the code is more readable with your proposal. I have requested some changed tho. 👍

**Comment by tillkruss** [comment]
Tests are failing.

**Comment by nunomaduro** [comment]
@tillkruss Yeah. There is a mistake on the new *if* condition 😕

**Comment by tillkruss** [comment]
Please resubmit with Ci passing.

File: src/Illuminate/Support/Arr.php (hunk 1 above)
Comment (rnaby) [reply src/Illuminate/Support/Arr.php]: Uh... I missed to turn the `and` to an `or`!

File: src/Illuminate/Support/Arr.php (hunk 2 above)
Comment (rnaby) [reply src/Illuminate/Support/Arr.php]: Uh... I missed to turn the `and` to an `or`!

**Pull request closed without merging**